### ISSUE TITLE
Find returns empty array but SQL query shows correct value

### ISSUE THREAD

**Issue by ullalaaron** [typeorm/typeorm#4121]
**Issue type:**

[* ] question
[* ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[* ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[* ] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

I am using this fantastic ORM with NestJS implementing the repository pattern.
I have a `LikeService` that has injected a `Repository` of my database `Like` entity (which has 3 foreign keys: `UserId`, `ContestId` and `PostId`)
In my `getPostIds()` method i run the following:

```
        return this.repository.find({
            where: {
                user: userId,
                contest: contestId,
            },
            select: ['post'],
        });
```

The executed query is the following:
`SELECT "Like"."postId" FROM "like" "Like" WHERE "Like"."userId" = $1 AND "Like"."contestId" = $2 -- PARAMETERS: [3,3]`
Which returns the following - desired - output:
![image](https://user-images.githubusercontent.com/17922843/57544242-16993880-7357-11e9-9f9b-91b74dbdc880.png)

The problem is that the above `getPostIds` method always returns an empty array. What am I missing?

**Comment by Kononnable** [comment]
Based on what you have provided - I have no clues. I might guess that something else is limiting your returned resources or you've forgot to await some promises.

**Comment by ikenami** [comment]
I'm having the same issue. Any updates?

In my case I have 2 TypeOrm connections. When I try to get the entries from the second connection (not the default), it returns nothing... Even though the database has the data I asked for.

**Comment by csenk** [comment]
We did find the issue today after multiple iterations. The issue was resolved as soon as the entity class name is exactly like the table name.

This will **NOT** work:
`
@Entity("foo_bar")
class FooBar extends BaseEntity {
}
`

This will work:
`
@Entity()
class foo_bar extends BaseEntity {
}
`

**Comment by haroldo-ok** [comment]
Same problem here, but none of the mentioned tricks has workded, so far.

**Comment by haroldo-ok** [comment]
Here's a piece of information which may, or may not, help  your case:

I'm working on adding support for Firebird dialect 1 support: https://github.com/haroldo-ok/typeorm-firebird

The main problem with FB dialect 1, as opposed to (dialect 3)[https://github.com/kerwynrg/typeorm/], is that there's no way to escape the table/colum/alias names, and all the aliases from the returned results are returned with uppercase.

I solved this partially by making the driver's `escape()` method uppercase everything; this partially solved the problem, except that "RawSqlResultsToEntityTransformer.group()" used "DriverUtils.buildColumnAlias()", which didn't do any kind of escaping; this made the grouping of the results fail, since it was expecting a different case.

Making "DriverUtils.buildColumnAlias()" call the driver's "escape()" method solved that problem, at least for FB dialect 1.

**Comment by jamesball27** [comment]
I ran into a similar issue in a project using the `typeorm-typedi-extensions` package. I wound up moving the connection options from the call to`createConnection` into a separate `ormconfig.json` file and that seemed to fix it 🤷

**Comment by JuanHB** [comment]
I had a similar issue as @ullalaaron , I'm loading my settings from the `ormconfig.js` file, and this file imports the settings from a `.env`... What was missing in my case, was to add the `schema` name in my. configs. 

```
...
const config: TypeOrmModuleOptions = {
  type,
  host: envOptions.TYPEORM_HOST,
  port: envOptions.TYPEORM_PORT,
  username: envOptions.TYPEORM_USERNAME,
  password: envOptions.TYPEORM_PASSWORD,
  database: envOptions.TYPEORM_DATABASE,
  schema: envOptions.TYPEORM_SCHEMA, // --> was missing this option
...
module.exports = config;
```

It was hard to debug in my case, because I was selecting from a table named `users` in my project's schema, but this table is present in other schemas, including the default `public`, which was the one TypeOrm was accessing.

I was only able to uncover this, because I enabled the logging by adding `logging: true` to the options object, which printed the query:
```
SELECT 
  "User"."id" AS "User_id", 
  "User"."name" AS "User_name", 
  "User"."email" AS "User_email", 
  "User"."password" AS "User_password", 
  "User"."status" AS "User_status", 
  "User"."created_at" AS "User_created_at", 
  "User"."updated_at" AS 
  "User_updated_at" 
FROM "users" "User"

SELECT * FROM users --> here is the problem, should have been: "MySchemaName"."users"
```

In conclusion, the problem was with my settings, TypeOrm was working just fine.

**Comment by imnotjames** [comment]
I think this is user error.  If it's still an issue please open a new issue with a reproducible example.

**Comment by imnotjames** [comment]
When opening an issue, people will be better able to provide help if you provide code that they can easily understand and use to reproduce the problem. This boils down to ensuring your code that reproduces the problem follows the following guidelines:

* Minimal – Use as little code as possible that still produces the same problem
* Complete – Provide all parts someone else needs to reproduce your problem in the question itself
* Reproducible – Test the code you're about to provide to make sure it reproduces the problem

## Minimal
The more code there is to go through, the less likely people can find your problem. Streamline your example in one of two ways:

1. Restart from scratch. Create a new program, adding in only what is needed to see the problem. Use simple, descriptive names for functions and variables – don’t copy the names you’re using in your existing code.
2. Divide and conquer. If you’re not sure what the source of the problem is, start removing code a bit at a time until the problem disappears – then add the last part back.

Don't sacrifice clarity for brevity when creating a minimal example. Use consistent naming and indentation, and include code comments if needed. Use your code editor’s shortcut for formatting code.

Don't include any passwords or credentials that must be kept secret.

## Complete

Make sure all information necessary to reproduce the problem is included in the issue itself.

If the problem requires some code as well as some XML-based configuration, include code for both. The problem might not be in the code that you think it is in.

Use individual code blocks for each file or snippet you include. Provide a description for the purpose of each block.

**DO NOT** use images of code. Copy the actual text from your code editor, paste it into the issus, then format it as code. This helps others more easily read and test your code.

## Reproducible

To help you solve your problem, others will need to verify that it exists.

Describe the problem. "It doesn't work" isn't descriptive enough to help people understand your problem. Instead, tell other readers what the expected behavior should be. Tell other readers what the exact wording of the error message is, and which line of code is producing it. Use a brief but descriptive summary of your problem as the title of your question.

Eliminate any issues that aren't relevant to the problem. If your question isn’t about a compiler error, ensure that there are no compile-time errors.

Double-check that your example reproduces the problem! If you inadvertently fixed the problem while composing the example but didn't test it again, you'd want to know that before asking someone else to help.
